tests/unit/test_utils.py

```diff
@@ -2598,6 +2598,24 @@ def test_can_use_127_ip_addr(self):
     def test_can_use_127_ip_addr_with_port(self):
         self.assert_can_retrieve_metadata_from('https://127.0.0.1:8080/foo')
 
+    def test_can_use_eks_ipv4_addr(self):
+        self.assert_can_retrieve_metadata_from(
+            'http://169.254.170.23/credentials'
+        )
+
+    def test_can_use_eks_ipv6_addr(self):
+        self.assert_can_retrieve_metadata_from(
+            'http://[fd00:ec2::23]/credentials'
+        )
+
+    def test_can_use_complex_uri(self):
+        self.assert_can_retrieve_metadata_from(
+            'http://127.0.0.1:8080/credentials?foo=bar%20baz'
+        )
```

Comment:
nit: for formatting we can alias the url so we only use two lines
```suggestion
        url = 'http://127.0.0.1:8080/credentials?foo=bar%20baz'
        self.assert_can_retrieve_metadata_from(url)
```